pyproject.toml

```diff
@@ -13,19 +13,22 @@ license = "MIT License"
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = ">=3.8.1,<3.12"
+python = ">=3.8.1,<3.11"
 loguru = "^0.6"
 numpy = "^1.23"
 pydantic = "^2.0"
 scipy = "^1.9"
 pymc = "^5.3"
 scikit-learn = "^1.1"
+obp = "^0.5.7"
```

Comment:
This library might be the reason why you had to downgrade python and pandas. 

At a glance, it looks not updated since years and anyway only supporting python up to 3.9

This library used to support python 3.12, and even downgrading it I am not sure about how it would behave with python 3.11 

I would evaluate if you can avoid to introduce it at all (this PR after all is small, you could try to rewrite the same code without that helper lib)

Reply:
You are indeed correct, obp is the reason for all the version downgrades. What should I do? Is it legit to "whiten" their code?

Reply:
Well, I would in principal try to avoid using that library at all. At the end here it is used in a single file. 
You can also try to open a PR there and bump their version. 

FYI I tried quickly, but I had some tests failing so it's probably not as straightforward

Reply:
![challenge-accepted-barney-stinson-06](https://github.com/user-attachments/assets/1812937f-1f9f-4223-b31a-013e57cd04ec)

Reply:
Hey @adarmiento,
Thank you for your valuable comments. I finished implementing all the OPE methods and am working on reviving python 3.12 for this repo.

Reply:
Great! 🚀

Reply:
And it's done. Python 3.12 is back on track. FYI @dariodandrea.